source/common/network/apple_dns_impl.cc

```diff
@@ -236,6 +264,8 @@ void AppleDnsResolverImpl::PendingResolution::onDNSServiceGetAddrInfoReply(
 
     if (!pending_cb_) {
       ENVOY_LOG(debug, "Adding to queries pending callback");
+      // DISCUSSION: should this be added on creation so even if a query returns success, but no new
+      // addresses the callbakc target is called?
```

Comment:
Sorry what is this case? You mean returns success with 0 results? Yes I think we should call the callback if there are no addresses.

Reply:
Yeah, exactly that case. Ok will change the logic and create a new test to cover.

Reply:
The only thing I'm not sure about here is what current code does with success with 0 addresses. I think it probably implicitly treats it as a failure but I might take a look at that and also what c-ares does in this case.

Reply:
c-ares can pass along a reply with Success and 0 addresses. Different targets do different things: the strict dns cluster [clears it host list](https://github.com/envoyproxy/envoy/pull/10137/files#diff-a0733fd5d5aec774af322c4de376c067ff5f75d3c16104985cb65495c8be044eR32) vs the logical dns cluster (and similar targets -- like the dns cache) [do not](https://github.com/envoyproxy/envoy/pull/10137/files#diff-a0733fd5d5aec774af322c4de376c067ff5f75d3c16104985cb65495c8be044eR61).

Reply:
OK sounds good then I would definitely do the same thing.